test/sqllogictest/role_create.slt

```diff
@@ -0,0 +1,148 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Loosely based on https://github.com/postgres/postgres/blob/master/src/test/regress/expected/create_role.out
+
+mode cockroach
+
+reset-server
+
+simple conn=mz_system,user=mz_system
+ALTER SYSTEM SET enable_ld_rbac_checks TO true;
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+ALTER SYSTEM SET enable_rbac_checks TO true;
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+CREATE ROLE regress_role_admin CREATEDB CREATECLUSTER CREATEROLE;
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+CREATE DATABASE regression
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+GRANT CREATE ON DATABASE regression TO regress_role_admin;
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+GRANT CREATE ON DATABASE regression TO regress_role_admin WITH GRANT OPTION;
+----
+db error: ERROR: Expected end of statement, found WITH
+
+simple conn=mz_system,user=mz_system
+CREATE ROLE regress_role_limited_admin CREATEROLE;
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+CREATE ROLE regress_role_normal;
+----
+COMPLETE 0
+
+simple conn=regress_role_limited_admin,user=regress_role_limited_admin
+CREATE ROLE regress_nosuch_superuser SUPERUSER;
+----
+db error: ERROR: SUPERUSER attribute is not supported, for more information consult the documentation at https://materialize.com/docs/sql/create-role/#details
+
+# This should fail, see https://github.com/MaterializeInc/materialize/issues/18960
+simple conn=regress_role_limited_admin,user=regress_role_limited_admin
+CREATE ROLE regress_nosuch_createdb CREATEDB;
+----
+COMPLETE 0
+
+# This should fail, see https://github.com/MaterializeInc/materialize/issues/18960
+simple conn=regress_role_limited_admin,user=regress_role_limited_admin
+CREATE ROLE regress_nosuch_createcluster CREATECLUSTER;
+----
+COMPLETE 0
+
+simple conn=regress_role_limited_admin,user=regress_role_limited_admin
+CREATE ROLE regress_role_limited;
+----
+COMPLETE 0
+
+# This should fail, see https://github.com/MaterializeInc/materialize/issues/18960
+simple conn=regress_role_limited_admin,user=regress_role_limited_admin
+ALTER ROLE regress_role_limited CREATEDB;
+----
+COMPLETE 0
+
+# This should fail, see https://github.com/MaterializeInc/materialize/issues/18960
+simple conn=regress_role_limited_admin,user=regress_role_limited_admin
+ALTER ROLE regress_role_limited CREATECLUSTER;
+----
+COMPLETE 0
+
+simple conn=regress_role_admin,user=regress_role_admin
+CREATE ROLE regress_createdb CREATEDB;
+----
+COMPLETE 0
+
+simple conn=regress_role_admin,user=regress_role_admin
+ALTER ROLE regress_createdb NOCREATEDB;
```

Comment:
For `ALTER ROLE` statements that legitimately should work, can we check that the new role properties are in effect both in the introspection table and by running a SQL query.

Note that `ALTER ROLE` is a bit different than `CREATE ROLE` in the sense that someone may have already established a connection using the role that is being modified. In this case, it would be implementation-defined (I do not know how Mz would do it) whether the changes to the role take place effective immediately, or only after a subsequent login with the role is attempted.